src/mon/OSDMonitor.cc

```diff
@@ -2135,6 +2135,11 @@ bool OSDMonitor::prepare_failure(MonOpRequestRef op)
 	dout(10) << " failure_info for osd." << target_osd << " now "
 		 << fi.reporters.size() << " reporters" << dendl;
       }
+      if (pending_inc.new_state.count(target_osd) &&
+          pending_inc.new_state[target_osd] & CEPH_OSD_UP) {
+        dout(10) << " cancel pending osd." << target_osd << " failure" << dendl;
+        pending_inc.new_state.erase(target_osd);
```

Comment:
There is a `pending_osd_state_clear` method which can do the same thing in a more safe and compact way.
E.g., we can avoid clearing another pending flags by accident.

Would you minding switching to use that method? Otherwise lgtm.

Reply:
@xiexingguo  switching done 
please review again, thanks